playwright_tests/E2E/case-flags.test.ts

```diff
@@ -0,0 +1,76 @@
+import { test, expect } from '@playwright/test';
+import { checkTableCellContent, checkTableRowContent, checkNumberOfRow } from "./steps/table-steps"
+import config from "../config"
+import { routeToCasePage } from './steps/case-steps';
+
+test('Create case flag 2', async ({ page }) => {
+  await loginExUIWithCaseFlag(page);
+  
+  await routeToCasePage(page, '1698182796208883');
+  await expect(page.getByText('Case flags', { exact: true })).toBeVisible();
+  await page.getByText('Case flags', { exact: true }).click();
+  await expect(page.getByRole('table', { name: 'Respondant' }).getByRole('caption')).toBeVisible();
+  await expect(page.locator('#case-viewer-field-read--FlagLauncher1').getByText('Language Interpreter')).toBeVisible();
+
+  console.log("Check Respondant details");
+  const tableClass= "govuk-table";
+  const tableName = "Respondant";
+  const rowTextRef = "Language Interpreter";
+  const textsToCheck = [/Test/, /Oct 2023/, /Active/];
+  await checkTableRowContent(page, tableClass, tableName, rowTextRef, textsToCheck);
+  await checkNumberOfRow(page, tableClass, tableName, 1+1);//One row for the header
+});
+
+test('View case flag', async ({ page }) => {
+  await loginExUIWithCaseFlag(page);
+  
+  await routeToCasePage(page, '1698182796208883');
+  
+  console.log("Check Applicant details");
+  await expect(page.getByText('There are 4 active flags on')).toBeVisible();
+  await page.getByText('Case flags', { exact: true }).click();
+  await expect(page.getByRole('table', { name: 'Applicant' }).getByRole('caption')).toBeVisible();
+  const tableClass= "govuk-table";
+  const tableName = "Applicant";
+  await checkNumberOfRow(page, tableClass, tableName, 5);//One row for the header
+  await expect(page.getByRole('table', { name: 'Applicant' }).getByRole('caption')).toBeVisible();
+  
+  const rowTextRef = "Support filling in forms";
+  const textsToCheck = [/Test/, /24 Oct 2023/, /Active/];
+  await checkTableRowContent(page, tableClass, tableName, rowTextRef, textsToCheck);
+
+  console.log("Check line 1");
+  await checkTableCellContent(page, tableName, 1, 0, "Documents in a specified colour");
```

Comment:
I think it would be worth writing a simple function to handle tables of content like:
interface TestDataRow{
  column: number,
  row: number,
  testData: string 
};
function checkTable(page: number, tableName: string, data: TestDataRow[]) {
  data.forEach (d => await checkTableCellContent(page, tableName, d.column, d.row, d.testData))
}

then you could replace n calls to checkTableCellContent with 

checkTable (page, tableName, [ 
{ 1, 1, "Test" },
{ 1, 2, "24 Oct 2023 }
 ...
 {2, 0, "Documents in a specified colour"}
etc
 }
It's not much effort, makes things a little more compact, and potentially allows for parallelisation of the checks in future. There's also potential to locate the tablename once and pass it in rather than fetching it every time, which should speed things up